### PR TITLE
BUGZ-692: Laucner is not movable & BUGZ-831 handle org and credentials erros.

### PR DESCRIPTION
**Pull request by danteruiz**
Tickets - https://highfidelity.atlassian.net/secure/RapidBoard.jspa?rapidView=37&modal=detail&selectedIssue=BUGZ-692, https://highfidelity.atlassian.net/secure/RapidBoard.jspa?rapidView=37&modal=detail&selectedIssue=BUGZ-831

### CODE REVIEW

**Comment by danteruiz** [comment]
Test plan: make sure that the launcher login process still works.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20183/HighFidelity-Beta-PR15854-6a5c909.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20183/HighFidelity-Beta-Interface-PR15854-6a5c909.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20183/HighFidelity-Beta-PR15854-6a5c909.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20183/HighFidelity-Beta-Interface-PR15854-6a5c909.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20183/HighFidelity-Beta-PR15854-6a5c909.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20183/HighFidelity-Quest-Beta-PR15854-6a5c909.apk)

File: launchers/darwin/src/CredentialsRequest.m [hunk 1]
```diff
@@ -21,77 +26,69 @@ - (void) confirmCredentials:(NSString*)username :(NSString*)password {
...
-                [sharedLauncher setTokenString:jsonString];
-                [sharedLauncher credentialsAccepted:TRUE];
-            });
-        }
-        
-        NSLog(@"credentials: connectionDidFinished completed");
-        
-    }];
-    
+    NSURLSession * session = [NSURLSession sessionWithConfiguration:NSURLSessionConfiguration.ephemeralSessionConfiguration delegate: self delegateQueue: [NSOperationQueue mainQueue]];
```
Comment (huffman) [inline launchers/darwin/src/CredentialsRequest.m]: Minor style issue: `*` has a space prepended

File: launchers/darwin/src/CredentialsRequest.m [hunk 2]
```diff
@@ -21,77 +26,69 @@ - (void) confirmCredentials:(NSString*)username :(NSString*)password {
...
+    if (error) {
+        NSLog(@"Credentials: Request completed with an error -> error: %@", error);
+        [sharedLauncher displayErrorPage];
     } else {
-        [[Settings sharedSettings] login:TRUE];
-        [sharedLauncher setTokenString:jsonString];
-        [sharedLauncher credentialsAccepted:TRUE];
+        if (self.statusCode == 200) {
+            NSString* jsonString = [[NSString alloc] initWithBytes: [self.receivedData mutableBytes] length:[self.receivedData length] encoding:NSUTF8StringEncoding];
+            NSData *jsonData = [jsonString dataUsingEncoding:NSUTF8StringEncoding];
```
Comment (huffman) [inline launchers/darwin/src/CredentialsRequest.m]: Minor style issue: `*` has a space prepended, and no following space

File: launchers/darwin/src/CredentialsRequest.m [hunk 3]
```diff
@@ -21,77 +26,69 @@ - (void) confirmCredentials:(NSString*)username :(NSString*)password {
...
+            NSError* jsonError = nil;
+            id json = [NSJSONSerialization JSONObjectWithData:jsonData options:0 error:&jsonError];
+
+            if (jsonError) {
+                NSLog(@"Credentials: Failed to parse json -> error: %@", jsonError);
```
Comment (huffman) [inline launchers/darwin/src/CredentialsRequest.m]: Do we print out the full response anywhere? It might be useful to print out here.

File: launchers/darwin/src/DownloadDomainContent.m [hunk 4]
```diff
@@ -36,18 +36,28 @@ -(void)URLSession:(NSURLSession *)session downloadTask:(NSURLSessionDownloadTask
...
+        [sharedLauncher displayErrorPage];
+        return;
+    }
     [sharedLauncher setDownloadContextFilename:destinationFileName];
     NSLog(@"extracting domain content file");
-    [sharedLauncher extractZipFileAtDestination:[[sharedLauncher getDownloadPathForContentAndScripts] stringByAppendingString:@"content"] :[[sharedLauncher getDownloadPathForContentAndScripts] stringByAppendingString:[sharedLauncher getDownloadContentFilename]]];
-    
+    BOOL extractionSuccessful = [sharedLauncher extractZipFileAtDestination:[[sharedLauncher getDownloadPathForContentAndScripts] stringByAppendingString:@"content"] :[[sharedLauncher getDownloadPathForContentAndScripts] stringByAppendingString:[sharedLauncher getDownloadContentFilename]]];
+
+    if (!extractionSuccessful) {
```
Comment (huffman) [inline launchers/darwin/src/DownloadDomainContent.m]: Can we log the error here?

File: launchers/darwin/src/DownloadInterface.m [hunk 5]
```diff
@@ -39,34 +39,44 @@ -(void)URLSession:(NSURLSession *)session downloadTask:(NSURLSessionDownloadTask
...
+    }
     [sharedLauncher setDownloadFilename:destinationFileName];
     NSString* appPath = [sharedLauncher getAppPath];
     NSString* downloadFileName = [sharedLauncher getDownloadFilename];
-    
+
     NSLog(@"extract interface zip");
-    [sharedLauncher extractZipFileAtDestination:appPath :[appPath stringByAppendingString:downloadFileName]];
+    BOOL success = [sharedLauncher extractZipFileAtDestination:appPath :[appPath stringByAppendingString:downloadFileName]];
+    if (!success) {
```
Comment (huffman) [inline launchers/darwin/src/DownloadInterface.m]: Can we log an error here?

File: launchers/darwin/src/OrganizationRequest.m [hunk 6]
```diff
@@ -23,70 +28,64 @@ - (void) confirmOrganization:(NSString*)aOrganization :(NSString*)aUsername {
...
-    [request setValue:@"application/json" forHTTPHeaderField:@"Content-Type"];
-
-    NSURLConnection *theConnection = [[NSURLConnection alloc] initWithRequest:request delegate:self];
+    [request setValue:@"" forHTTPHeaderField:@"Content-Type"];
 
-    if(theConnection) {
-        self.webData = [NSMutableData data];
-        NSLog(@"connection initiated");
-    }*/
+    NSURLSession * session = [NSURLSession sessionWithConfiguration:NSURLSessionConfiguration.ephemeralSessionConfiguration delegate: self delegateQueue: [NSOperationQueue mainQueue]];
```
Comment (huffman) [inline launchers/darwin/src/OrganizationRequest.m]: Minor style issue: `*` has a space prepended

File: launchers/darwin/src/OrganizationRequest.m [hunk 7]
```diff
@@ -23,70 +28,64 @@ - (void) confirmOrganization:(NSString*)aOrganization :(NSString*)aUsername {
...
+        if (self.statusCode == 200) {
+            NSError *jsonError = nil;
+            NSDictionary *json = [NSJSONSerialization JSONObjectWithData:self.receivedData options:kNilOptions error:&jsonError];
+            if (jsonError) {
+                NSLog(@"Failed to parse Organzation json -> error: %@", jsonError);
+                [sharedLauncher displayErrorPage];
+                return;
+            }
+            NSLog(@"Organization: getting org file successful");
+            [sharedLauncher setDomainURLInfo:[json valueForKey:@"domain"] :[json valueForKey:@"content_set_url"] :[json valueForKey:@"scripts_url"]];
```
Comment (huffman) [inline launchers/darwin/src/OrganizationRequest.m]: What happens if these keys don't exist? Can we handle that error?

File: launchers/darwin/src/DownloadDomainContent.m (hunk 4 above)
Comment (danteruiz) [reply launchers/darwin/src/DownloadDomainContent.m]: the error is logged in the function above.

File: launchers/darwin/src/DownloadInterface.m (hunk 5 above)
Comment (danteruiz) [reply launchers/darwin/src/DownloadInterface.m]: the error is logged in the function above.

File: launchers/darwin/src/OrganizationRequest.m (hunk 7 above)
Comment (danteruiz) [reply launchers/darwin/src/OrganizationRequest.m]: I could do that.

File: launchers/darwin/nib/DisplayNameScreen.xib [hunk 8]
```diff
@@ -74,8 +74,8 @@
                     <autoresizingMask key="autoresizingMask"/>
                     <textFieldCell key="cell" lineBreakMode="clipping" alignment="center" title="By signing in, you agree to the High Fidelity" id="zSf-YA-osu">
                         <font key="font" metaFont="system"/>
-                        <color key="textColor" name="labelColor" catalog="System" colorSpace="catalog"/>
-                        <color key="backgroundColor" name="textBackgroundColor" catalog="System" colorSpace="catalog"/>
+                        <color key="textColor" name="alternateSelectedControlTextColor" catalog="System" colorSpace="catalog"/>
+                        <color key="backgroundColor" name="alternateSelectedControlTextColor" catalog="System" colorSpace="catalog"/>
```
Comment (huffman) [inline launchers/darwin/nib/DisplayNameScreen.xib]: Were these changes intentional?

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20192/HighFidelity-Beta-PR15854-12477f3.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20192/HighFidelity-Beta-Interface-PR15854-12477f3.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20192/HighFidelity-Beta-PR15854-12477f3.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20192/HighFidelity-Beta-Interface-PR15854-12477f3.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20192/HighFidelity-Beta-PR15854-12477f3.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20192/HighFidelity-Quest-Beta-PR15854-12477f3.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20194/HighFidelity-Beta-PR15854-12477f3.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20194/HighFidelity-Beta-Interface-PR15854-12477f3.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20194/HighFidelity-Beta-PR15854-12477f3.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20194/HighFidelity-Quest-Beta-PR15854-12477f3.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20194/HighFidelity-Beta-PR15854-12477f3.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20194/HighFidelity-Beta-Interface-PR15854-12477f3.exe
